From 42b0b5adb9154519a2eda7da81a993c17a536f0a Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Sat, 17 Mar 2018 17:02:39 +0100
Subject: [PATCH] Unset QT_QPA_PLATFORM to get xcb

Currently we can't get wayland applications to work, as they rely on
XDG_RUNTIME_DIR being set.
With wayland 1.15, WAYLAND_DISPLAY can be an absolute path though, so this
can be replaced by setting WAYLAND_DISPLAY=$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY
before unsetting XDG_RUNTIME_DIR and removing the unsetenv of QT_QPA_PLATFORM.
---
 src/kdesu_stub.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/kdesu_stub.c b/src/kdesu_stub.c
index 49db577..5038628 100644
--- a/src/kdesu_stub.c
+++ b/src/kdesu_stub.c
@@ -315,6 +315,9 @@ int main()
     /* Handle display */
 
     if (strcmp(params[P_DISPLAY].value, "no")) {
+        /* We only handle X11 - so unset QT_QPA_PLATFORM to get xcb. */
+        unsetenv("QT_QPA_PLATFORM");
+
         xsetenv("DISPLAY", params[P_DISPLAY].value);
         if (params[P_DISPLAY_AUTH].value[0]) {
             int fd2;
-- 
2.16.2

